examples/example_composed_target_function.py

```diff
@@ -0,0 +1,160 @@
+"""
+Example for evolutionary regression on a composed target function
+==================================================================
+
+Example demonstrating the use of Cartesian genetic programming for
+regression on a target function composed of two different sub-functions.
+Demonstrates the use of comparison operator genes (>, <) and if/else branching genes.
+"""
+
+import functools
+import numpy as np
+
+import cgp
+
+
+# Define new operator nodes
+class ConstantFloatZero(cgp.ConstantFloat):
+    # Should be useful for comparison operator (given the target function structure)
+    def __init__(self, idx, inputs):
+        super().__init__(idx, inputs)
+        self._output = 0
+
+
+class IfElseOperator(cgp.OperatorNode):
+    """A node that checks its input0 is bigger than or equal 0
+     and outputs input1 if true and input2 otherwise """
+
+    _arity = 3
+
+    def __init__(self, idx, inputs):
+        super().__init__(idx, inputs)
+        if inputs[0] >= 0:
+            self._output = inputs[1]
+        else:
+            self._output = inputs[2]
+
+    _def_output = "x_1 if x_0 >= 0 else x_2"
+    _def_numpy_output = "np.piecewise(x_0, [x_0>= 0, x_0<0], [x_1 [x_0>= 0] , x_2[x_0<0]])"
+    _def_sympy_output = "Piecewise((x_1, x_0 >= 0), (x_2, x_0 < 0))"
+    _def_torch_output = "torch.where(x_0 >= 0, x_1, x_2)"
+
+
+class ComparisonOperator(cgp.OperatorNode):
+    """A node that compares its inputs and outputs 1.0 if the first is bigger,
+     0 if they are equal and - 1.0 if the second is bigger """
+
+    _arity = 2
+    _def_output = "1.0 * ((x_0 - x_1)/max(abs(x_0 - x_1), 0.000001))"
+    _def_sympy_output = "1.0 * sign(x_0 - x_1)"
+    _def_torch_output = "1.0 * torch.sign(x_0 - x_1)"
+    _def_numpy_output = "1.0 * np.sign(x_0 - x_1)"
+
+
+def f_target(x):
+    y = np.zeros_like(x)
+    y[x <= 0] = x[x <= 0] ** 2
+    y[x > 0] = x[x > 0]
```

Comment:
In principle, we're solving the same task as in the example that I added in #230 . I would propose to use the same target function as in that example and keep everything as close as possible to that example, to make it easier to compare both examples.